source/mir/random/mixture.d

```diff
@@ -0,0 +1,109 @@
+/**
+Mixture methods that can be used to combine random variables.
+
+License: $(LINK2 http://boost.org/LICENSE_1_0.txt, Boost License 1.0).
+
+Authors: Ilya Yaroshenko, Sebastian Wilzbach
+*/
+
+module mir.random.mixture;
+
+import std.traits : isNumeric;
+
+/**
+_Discrete distribution sampler.
+Given an array of cumulative density points `cdPoints`,
+a _discrete distribution is defined.
+The cumulative density points can be calculated from probabilities or counts
+using `std.algorithm.cumulativeFold`. Hence `cdPoints` is assumed to be sorted.
+
+Params:
+    cdPoints = cumulative density points
+
+Returns:
+    A sampler that can be called to sample from the distribution.
+*/
+struct Discrete(T)
+    if (isNumeric!T)
```

Comment:
> Struct declaration should be after function. And add a small separate comment please

ok for the order, but `ditto` would group them:

![image](https://cloud.githubusercontent.com/assets/4370550/16361404/35bde8ca-3b90-11e6-9126-c75c0104dcae.png)